model/Index/Handler/ResourceUpdatedHandler.php

```diff
@@ -0,0 +1,195 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2022 (original work) Open Assessment Technologies SA.
+ */
+
+namespace oat\taoAdvancedSearch\model\Index\Handler;
+
+use common_Exception;
+use common_exception_InconsistentData;
+use core_kernel_classes_Resource;
+use Exception;
+use oat\generis\model\data\event\ResourceUpdated;
+use oat\oatbox\event\Event;
+use oat\tao\model\search\index\DocumentBuilder\IndexDocumentBuilderInterface;
+use oat\tao\model\search\index\IndexDocument;
+use oat\tao\model\search\SearchInterface;
+use oat\taoAdvancedSearch\model\Index\Specification\ItemResourceSpecification;
+use oat\taoItems\model\media\ItemMediaResolver;
+use oat\taoQtiItem\model\qti\interaction\MediaInteraction;
+use oat\taoQtiItem\model\qti\Service as QtiService;
+use Psr\Log\LoggerInterface;
+use tao_helpers_Uri;
+use Throwable;
+
+class ResourceUpdatedHandler extends AbstractEventHandler
```

Comment:
Lets just double check if we can rely on `ItemUpdatedEvent` or we can have a generic one, but not harm the performance.

Reply:
Right now determining the type of resource is delegated to a new class (`ResourceReferencesService`), since we need that logic also in other places (i.e. `UpdateResourceInIndex` task from TAO Core, [here](https://github.com/oat-sa/tao-core/pull/3673/files#diff-41f44ebd716e148a06deb3815dfa01c550963f9412f196da5b7a50ee581d99e5R115)). We need to determine the type of resource to pick a method to extract the referenced resources' URIs.

That means the implementation for `TestUpdatedHandler` and `ResourceUpdatedHandler` is more-or-less the same: They only differ in how they get the actual resource URI from the event (as the events they handle expose it under different attribute names), and then they call the new `ResourceReferencesService` class.

So, maybe we can merge both listeners into a single one that would have some additional logic to check the particular type of event we've received and use the appropriate method to extract the resource URI. That way we could drop one of the two listeners.

Please note that, for me (locally), `ItemUpdatedEvent` is triggered first, then `ResourceUpdated`. That last one ends up calling `ResourceWatcher::catchUpdatedResourceEvent()`, which in turn may spawn a new `UpdateResourceInIndex` task. If that happens, the task uses the same logic from `ResourceReferencesService` in order to generate a document for the item that also contains the asset references.